Formula/aircrack-ng.rb

```diff
@@ -34,6 +34,8 @@ def install
                            "--disable-dependency-tracking",
                            "--prefix=#{prefix}",
                            "--with-experimental"
+    # On MacOS X, some of the c++ header files #include <version> and the makefile has -I. causing problems
+    system "mv", "version", "version.txt"
```

Comment:
Can we do an out-of-tree build instead of messing around with source files that might be used in the build?

Reply:
Try
```diff
diff --git a/Formula/aircrack-ng.rb b/Formula/aircrack-ng.rb
index 0df8c8792ad..1a7a35b5cb2 100644
--- a/Formula/aircrack-ng.rb
+++ b/Formula/aircrack-ng.rb
@@ -30,11 +30,13 @@ class AircrackNg < Formula
   patch :DATA
 
   def install
-    system "./autogen.sh", "--disable-silent-rules",
-                           "--disable-dependency-tracking",
-                           "--prefix=#{prefix}",
-                           "--with-experimental"
-    system "make", "install"
+    mkdir "build" do
+      system "../autogen.sh", "--disable-silent-rules",
+                              "--disable-dependency-tracking",
+                              "--prefix=#{prefix}",
+                              "--with-experimental"
+      system "make", "install"
+    end
   end
```

Reply:
I fixed this to use ruby mv, and on_macos, but I'm not sure the best way to do an out of tree build, I'll play with how aircrack works.

Reply:
```
diff --git a/Formula/aircrack-ng.rb b/Formula/aircrack-ng.rb
index 6871d2d06ed..8b7ce361846 100644
--- a/Formula/aircrack-ng.rb
+++ b/Formula/aircrack-ng.rb
@@ -30,15 +30,17 @@ class AircrackNg < Formula
   patch :DATA
 
   def install
-    system "./autogen.sh", "--disable-silent-rules",
-                           "--disable-dependency-tracking",
-                           "--prefix=#{prefix}",
-                           "--with-experimental"
-    # On MacOS X, some of the c++ header files #include <version> and the makefile has -I. causing problems
-    on_macos do
-      mv "version", "version.txt"
+    # build out-of-tree to avoid header collision with
+    # the version file in source root
+    mkdir "build" do
+      system "../autogen.sh", "--disable-silent-rules",
+                                "--disable-dependency-tracking",
+                                "--prefix=#{prefix}",
+                                "--with-experimental"
+      # On MacOS X, some of the c++ header files #include <version> and the makefile has -I. causing problems
+      #mv "version", "version.txt"
+      system "make", "install"
     end
-    system "make", "install"
   end
 
   def caveats
```

It's not liking that, I'm going to have to futz with it tomorrow.

Reply:
interesting.  I think if I build this on a case-sensitive file system the problem will go away.
it's VERSION in the tree.

Reply:
It's very much possible that out-of-tree builds don't help at all: https://github.com/aircrack-ng/aircrack-ng/blob/master/Makefile.am#L47

If not, I wouldn't worry about it really.